js/codemeta_generation.js

```diff
@@ -115,49 +128,65 @@ function generatePersons(prefix) {
     return persons;
 }
 
-function generateCodemeta() {
-    var inputForm = document.querySelector('#inputForm');
-    var codemetaText, errorHTML;
 
-    if (inputForm.checkValidity()) {
-        var doc = {
-            "@context": "https://doi.org/10.5063/schema/codemeta-2.0",
-            "@type": "SoftwareSourceCode",
-        };
-
-        let licenses = getLicenses();
-        if (licenses.length > 0) {
-            doc["license"] = (licenses.length === 1) ? licenses[0] : licenses;
-        }
+function buildDoc() {
+    var doc = {
+        "@context": CODEMETA_CONTEXT_URL,
+        "@type": "SoftwareSourceCode",
+    };
 
-        // Generate most fields
-        directCodemetaFields.forEach(function (item, index) {
-            doc[item] = getIfSet('#' + item)
-        });
+    let licenses = getLicenses();
+    if (licenses.length > 0) {
+        doc["license"] = (licenses.length === 1) ? licenses[0] : licenses;
+    }
```

Comment:
```suggestion
    doc["license"] = getLicenses();
```

This should be equivalent given that we run the compaction algorithm later on. (and then you can fold it in the `var doc =` expression above.)

Reply:
I do have to check for the existence of licenses though, otherwise an empty field `licences: []` is kept in the generated codemeta, which breaks the tests.

Reply:
Ah, indeed